src/placement/tests/jump_map_place_obj.c

```diff
@@ -24,553 +24,1735 @@
 
 #include <daos/common.h>
 #include <daos/placement.h>
-#include "daos_api.h"
 #include <daos.h>
 #include "place_obj_common.h"
+/* Gain some internal knowledge of pool server */
+#include "../../pool/rpc.h"
+#include "../../pool/srv_pool_map.h"
+
+#include <stdarg.h>
+#include <stddef.h>
+#include <setjmp.h>
+#include <cmocka.h>
+#include <daos/tests_lib.h>
+#define STR_HELPER(x) #x
+#define STR(x) STR_HELPER(x)
+
+static bool g_verbose;
+
+#define skip_msg(msg) do { print_message(__FILE__":" STR(__LINE__) \
+			" Skipping > "msg"\n"); skip(); } \
+			while (0)
+#define is_true assert_true
+#define is_false assert_false
+
+void verbose_msg(char *msg, ...)
+{
+	if (g_verbose) {
+		va_list vargs;
 
-#define DOM_NR		18
-#define	NODE_PER_DOM	1
-#define VOS_PER_TARGET	4
-#define SPARE_MAX_NUM	(DOM_NR * 3)
-#define COMPONENT_NR	(DOM_NR + DOM_NR * NODE_PER_DOM + \
-			 DOM_NR * NODE_PER_DOM * VOS_PER_TARGET)
-#define NUM_TARGETS	(DOM_NR * NODE_PER_DOM * VOS_PER_TARGET)
+		va_start(vargs, msg);
+		vprint_message(msg, vargs);
+		va_end(vargs);
+	}
+}
 
-#define TEST_PER_OC 1000
+static void
+gen_maps(int num_domains, int nodes_per_domain, int vos_per_target,
+	 struct pool_map **po_map, struct pl_map **pl_map)
+{
+	*po_map = NULL;
+	*pl_map = NULL;
+	gen_pool_and_placement_map(num_domains, nodes_per_domain,
+				   vos_per_target, PL_TYPE_JUMP_MAP,
+				   po_map, pl_map);
+	assert_non_null(*po_map);
+	assert_non_null(*pl_map);
+}
 
-static bool			 pl_debug_msg;
+static void
+gen_oid(daos_obj_id_t *oid, uint64_t lo, uint64_t hi, daos_oclass_id_t cid)
+{
+	oid->lo = lo;
+	/* make sure top 32 bits are unset (DAOS only) */
+	oid->hi = hi & 0xFFFFFFFF;
+	daos_obj_generate_id(oid, 0, cid, 0);
+}
 
-void
-placement_object_class(daos_oclass_id_t cid)
+#define assert_placement_success(pl_map, cid) \
+	do {\
+		daos_obj_id_t __oid; \
+		struct pl_obj_layout *__layout = NULL; \
+		gen_oid(&__oid, 1, UINT64_MAX, cid); \
+		assert_success(plt_obj_place(__oid, &__layout, pl_map, \
+				false)); \
+		pl_obj_layout_free(__layout); \
+	} while (0)
+#define assert_invalid_param(pl_map, cid) \
+	do {\
+		daos_obj_id_t __oid; \
+		struct pl_obj_layout *__layout = NULL; \
+		gen_oid(&__oid, 1, UINT64_MAX, cid); \
+		assert_err(plt_obj_place(__oid, &__layout, pl_map, \
+				false), -DER_INVAL); \
+	} while (0)
+
+static void
+object_class_is_verified(void **state)
 {
 	struct pool_map		*po_map;
 	struct pl_map		*pl_map;
-	struct pl_obj_layout	*layout;
-	daos_obj_id_t		oid;
-	int			test_num;
+#define DAOS_6295 0
```

Comment:
Why is this defined as 0, and not 1?

Reply:
Because it's defined as 0, the #if DAOS-6295 checks will fail and those parts of the test won't run (because they fail not anyway). Probably a better way to have done this, but the idea was that when working on DAOS-6295, that define could be set to 1 and expose the failing parts.